### PR TITLE
docs: improve typecheck FAQ

### PR DESCRIPTION
**Pull request by ldez**
rephrases the doc to try to be more clear about the nature of `typecheck`.

### CODE REVIEW

File: docs/src/docs/usage/faq.mdx [hunk 1]
```diff
@@ -35,15 +35,22 @@ Usually this options is used during development on local machine and compilation
...
 
 It cannot be disabled because of that.
 
 Of course, this is just as good as the compiler itself and a lot of compilation issues will not properly show where in the code your error lies.
 
-`typecheck` is not a real linter, it's just a way to parse compiling errors (produced by the `types.Checker`) and some linter errors.
+As consequence, the code to analyze should compile.
+It means that if you try to run an analysis on a single file or a group of files or a package or a group of packages,
+with dependencies on other files or packages of your project, as it doesn't compile (because of the missing pieces of code),
+it also cannot be analyzed.
```
Comment (bombsimon) [inline docs/src/docs/usage/faq.mdx:51]: Should a short version of this be added to the troubleshooting list to clarify that not only try `go build ./...` but also try `golangci-lint run ./...`?

**Review by bombsimon — APPROVED** [review]
LGTM!

As suggested in #4303 I think this can be added to the list to make it even clearer what to try before filing an issue.

File: docs/src/docs/usage/faq.mdx (hunk 1 above)
Comment (ldez) [reply docs/src/docs/usage/faq.mdx:51]: It's complex to summarize this sentence because each part is important to avoid ambiguity.

**Review by bombsimon — APPROVED** [review]

**Comment by nuhasha** [comment]
@ldez I have a question: how do I run the analysis? based on what you mentioned: 

> Ensure you are not running an analysis on code that depends on files/packages outside the scope of the analyzed elements.

for example in my PR it gives an error on one of the variables in `main.go`

``` 
Error: main.go:153:7: cfg.variableX undefined (type Config has no field or method variableX) (typecheck)
  		
```
but `variableX` is defined in `folder1/folder2/config.go`

the linting file is part of GitHub workflows `.github/workflows/linting.yaml ` so, I am not sure what I am doing wrong here.

Here is my repo structure
```
|main.go
|folder1
|    folder2
|        config.go
|.github
|     workflows
|          linting.yaml  
```

**Comment by ldez** [comment]
@nuhasha can you open a discussion? https://github.com/golangci/golangci-lint/discussions/categories/q-a

**Pull request merged**